### ISSUE TITLE
slice fails in tt-sim

### ISSUE THREAD

**Issue by bbradelTT** [tenstorrent/tt-metal#38246]
Slice fails on tt-sim. See https://github.com/tenstorrent/tt-metal/issues/38232
```
2026-02-20 17:18:36.270 | debug    |              Op | Launching Device Operation: "SliceDeviceOperation" (device_operation.hpp:151)
2026-02-20 17:18:36.270 | debug    |              Op | Program Hash: 18446744073332484884 (device_operation.hpp:153)
2026-02-20 17:18:36.270 | debug    |              Op | Program Cache Hit: false (device_operation.hpp:154)
2026-02-20 17:18:36.270 | debug    |              Op | Attributes: (device_operation.hpp:156)
2026-02-20 17:18:36.270 | debug    |              Op | 	slice_start = Shape([0, 0]) (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | 	slice_end = Shape([0, 0]) (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | 	step = Shape([1, 1]) (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | 	output_mem_config = MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt,nd_shard_spec=std::nullopt,created_with_nd_shard_spec=0) (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | 	use_tensor_args = true (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | 	slice_dim = 1 (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | 	num_devices = 1 (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | 	sub_core_grids = std::nullopt (device_operation.hpp:158)
2026-02-20 17:18:36.270 | debug    |              Op | Tensors Args: (device_operation.hpp:161)
2026-02-20 17:18:36.270 | debug    |              Op | 	0: Tensor(storage=DeviceStorage(),tensor_spec=TensorSpec(logical_shape=Shape([32, 128256]),tensor_layout=TensorLayout(dtype=DataType::INT32,page_config=PageConfig(config=TilePageConfig(tile=Tile(tile_shape={32, 32},face_shape={16, 16},num_faces=4))),memory_config=MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt,nd_shard_spec=std::nullopt,created_with_nd_shard_spec=0),alignment=Alignment([32, 32])))) (device_operation.hpp:165)
2026-02-20 17:18:36.270 | debug    |              Op | 	1: Tensor(storage=DeviceStorage(),tensor_spec=TensorSpec(logical_shape=Shape([2]),tensor_layout=TensorLayout(dtype=DataType::INT32,page_config=PageConfig(config=RowMajorPageConfig(tile=Tile(tile_shape={32, 32},face_shape={16, 16},num_faces=4))),memory_config=MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt,nd_shard_spec=std::nullopt,created_with_nd_shard_spec=0),alignment=Alignment([1])))) (device_operation.hpp:165)
2026-02-20 17:18:36.270 | debug    |              Op | 	2: Tensor(storage=DeviceStorage(),tensor_spec=TensorSpec(logical_shape=Shape([2]),tensor_layout=TensorLayout(dtype=DataType::INT32,page_config=PageConfig(config=RowMajorPageConfig(tile=Tile(tile_shape={32, 32},face_shape={16, 16},num_faces=4))),memory_config=MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt,nd_shard_spec=std::nullopt,created_with_nd_shard_spec=0),alignment=Alignment([1])))) (device_operation.hpp:165)
2026-02-20 17:18:36.270 | debug    |              Op |  (device_operation.hpp:170)
[654047] ERROR: UndefinedBehavior: tensix_execute_unpacr: unpack_to_dst=0 in_data_format=8
```

**Comment by nkapreTT** [comment]
### Minimal repro: `to_layout(sub_core_grids=...)` tilize zeros the first output tile on tt-sim

Surfaced this while running the Blackhole TTNN post-commit op set through the craq-sim/tt-sim functional simulator (it makes every `matmul/test_linear.py::test_linear_on_subdevice*` case fail). Root-caused it down to a tilize defect that needs **no matmul, no sharding, no multicast, no sub-device offset, and a single core** — which looks like the same UNPACR-tilize family as this issue.

**Symptom:** a `sub_core_grids` device tilize (`to_layout(..., TILE_LAYOUT, sub_core_grids=...)`) returns its **first 32×32 output tile as all-zero**; every other tile is **bit-exact**. Silent — no assert, no warning.

**Minimal repro (single core, 2 tiles):**
```python
import torch, ttnn
def test_subcore_tilize_first_tile_zero(device):
    crs = ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(0, 0))})
    b = torch.randn((32, 64), dtype=torch.bfloat16)            # one tile-row, two tile-cols
    ib = ttnn.to_layout(ttnn.from_torch(b, dtype=ttnn.bfloat16, device=device),
                        ttnn.TILE_LAYOUT, sub_core_grids=crs)
    rt = ttnn.to_torch(ib).reshape(32, 64).to(torch.float32)
    bf = b.to(torch.float32)
    # tile-col 0 (cols 0..31) comes back all-zero; tile-col 1 (cols 32..63) is bit-exact
    assert rt[:, 0:32].abs().sum() == 0          # BUG: first tile is zeroed
    assert torch.equal(rt[:, 32:], bf[:, 32:])   # rest is correct
```

Observed:
```
b[0,0:8]    = [1.46, 0.29, 0.96, 1.57, 0.49, -1.52, -0.25, 0.57]
rt[0,0:8]   = [0,    0,    0,    0,    0,    0,     0,    0  ]   # first tile zeroed
rt[0,32:40] = b[0,32:40]                                         # bit-exact
```

**Confirmed invariant** across 32×64, 64×64, and 512×512 on a single core (always exactly the first output tile), and it propagates through `ttnn.linear` on a `sub_core_grids`/sub-device grid as a corrupted first output N-tile → ~PCC 0.94 (≈1/16 of the output wrong). Whether the defect is in the sim's UNPACR/tilize datapath (this issue) vs the host `sub_core_grids` tilize program (cf. #21369) hasn't been separated without silicon; the **silent zero (no crash)** points at the sim UB here.

Backend: craq-sim/tt-sim functional simulator (BH, unharvested `blackhole_140` descriptor), regular interleaved `to_layout`/tilize works fine — only the `sub_core_grids` path zeros the first tile.

**Comment by franciseliang99-dot** [comment]
A data point that may be relevant to the LLK tag removed on 2026-04-06 ("Not an LLK issue").

I filed the same root cause against the simulator as tenstorrent/ttsim#17 (Wormhole, ttsim 1.10.3). It was closed on 2026-09-14 with the following reasoning: per [`UNPACR_Regular.md`](https://github.com/tenstorrent/tt-isa-documentation/blob/main/WormholeB0/TensixTile/TensixCoprocessor/UNPACR_Regular.md), FP32/INT32 with `!UnpackToDst` is intentionally `UndefinedBehavior`:

```
// FP32/INT32 with !UnpackToDst is intentionally UndefinedBehavior. SrcA/SrcB datum slots are 19 bits;
// ... the exact behavior has not been validated and must not be relied upon by software.
case INT32: return UnpackToDst ? DstEncodeFP32(DatumBits) : UndefinedBehavior();
case FP32:  return UnpackToDst ? DstEncodeFP32(DatumBits) : UndefinedBehavior();
```

and "is therefore not legal SW behavior". So on the simulator side this is treated as a software issue, not a sim gap.

The instruction in question is still on `main`: the TEN-3868 workaround in `unpack_to_dest_tile_done()`, which @iwroszTT traced above:

https://github.com/tenstorrent/tt-metal/blob/99da4d79707583384694461b93699d78d160cb8b/tt_metal/tt-llk/tt_llk_wormhole_b0/common/inc/cunpack_common.h#L987-L989

The fix PR mentioned above, tenstorrent/tt-llk#1357 ("unpack workaround to use data format valid for Src reg"), was closed without merging on 2026-06-07.

Scope on Wormhole, in case it helps with prioritisation. This is from a full tt-llk pytest run against ttsim 1.10.3, on tt-metal `10c2e67eff` (2026-09-08), using `run_ttsim_regression.sh -n 6 -a wormhole -t 600`:

- 41,670 cases in total, 1,554 failures.
- 1,477 of those failures are this `tensix_unpacr: unpack_to_dst=0` UB. It shows up with in/out format 0 (Float32) in 854 cases and with in format 8 (Int32/UInt32) in 623 cases, across 20 test files.
- Bisect over `test_bcast.py test_transpose_dest.py test_pack_rows.py test_unpack_tilize.py`, with only that `TT_UNPACR` commented out: `154 failed / 254 passed` → `408 passed / 0 failed`. This was a diagnostic run only. I'm not proposing removing the instruction, since that would re-expose TEN-3868 on silicon.
- Independent split: `test_transpose_dest.py` has `unpack_to_dest` as an axis in its test ids. `False` gives 18 passed / 0 failed, and `True` gives 0 passed / 10 failed.

Limits of what I know:
- I have no silicon, so I can't say what hardware does with this sequence. I'm relying only on the ISA doc and the ttsim maintainer's ruling.
- @bbradelTT noted above that removing the UB surfaced `UntestedFunctionality: tensix_execute_unpacr: zero_write2=1` next. This workaround UNPACR also sets ZeroWrite2, so a fix that only changes the data format (the #1357 approach) may still trip that check on ttsim. I haven't tested that.

If a candidate branch exists, I'm happy to rerun the Wormhole ttsim sweep against it and post before/after numbers.
